scripts/asm-installer/install_asm

```diff
@@ -1397,13 +1398,33 @@ EOF
 
   info "Downloading ASM kpt package..."
   retry 3 kpt pkg get --auto-set=false "${KPT_URL}" asm
+  version_message > "${ASM_VERSION_FILE}"
 }
 
 necessary_files_exist() {
   if [[ ! -f "${OUTPUT_DIR}/${ISTIOCTL_REL_PATH}" ]]; then
     false
+    return
```

Comment:
This alone will solve a bunch of issues.

Reply:
That by itself won't change any of the behavior. There was nothing after the if block to execute so it would return anyways.